cluster.go

```diff
@@ -207,15 +207,26 @@ func (c *clusterClient) _refresh() (err error) {
 
 	groups := result.parse(c.opt.TLSConfig != nil)
 	conns := make(map[string]connrole, len(groups))
-	for master, g := range groups {
-		conns[master] = connrole{conn: c.connFn(master, c.opt)}
+	for primary, g := range groups {
+		primaryConn := c.connFn(primary, c.opt)
+		conns[primary] = connrole{conn: primaryConn}
+		if len(g.nodes) > 0 {
+			g.nodes[0].AZ = primaryConn.AZ()
```

Comment:
I think the AZ() is not yet available right after `c.connFn`. `c.connFn` only constructs the `*mux` without making the actual TCP connection. We may need another way to do this.

Reply:
Oh, I just saw you implemented AZ() as `m.pipe(0).AZ()` then it would make an actual connection to the server.
But here is still not a good place to call AZ(). It will be better we call AZ() only when the AZ_AFFINITY is enable and call it here https://github.com/redis/rueidis/blob/09ad427372a36519b3ef64c6678d755e749b51ef/cluster.go#L265 with batching.

Reply:
So, you mean not always set AZ to `ReplicaInfo` but if user want to know, then set AZ to `ReplicaInfo`?

Reply:
I mean the `AZ()` should be invoked only when necessary, which is when the AZ_AFFINITY is configured.

Reply:
And, the current calling point, which is right after `result.parse`, is not a good place to call `AZ()`. It will always create new connections. We should instead call it after:
https://github.com/redis/rueidis/blob/09ad427372a36519b3ef64c6678d755e749b51ef/cluster.go#L232-L243

So that we can reuse existing connections as much as possible.